### ISSUE TITLE
HMat prevents OT from building

### ISSUE THREAD

**Issue by josephmure** [openturns/openturns#1386]
HMat prevents OT from building the master:

```
[  3%] Building CXX object lib/src/CMakeFiles/OT.dir/Base/Stat/ProcessSample.cxx.o
/home/osboxes/GIT/openturns/lib/src/Base/Stat/HMatrixImplementation.cxx: In member function ‘OT::Bool OT::HMatrixImplementation::setKey(const String&, const String&)’:
/home/osboxes/GIT/openturns/lib/src/Base/Stat/HMatrixImplementation.cxx:543:36: error: ‘hmat_compress_aca_random’ was not declared in this scope
       settings.compressionMethod = hmat_compress_aca_random;
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
/home/osboxes/GIT/openturns/lib/src/Base/Stat/HMatrixImplementation.cxx:543:36: note: suggested alternative: ‘hmat_compress_aca_plus’
       settings.compressionMethod = hmat_compress_aca_random;
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
                                    hmat_compress_aca_plus
lib/src/CMakeFiles/OT.dir/build.make:7646: recipe for target 'lib/src/CMakeFiles/OT.dir/Base/Stat/HMatrixImplementation.cxx.o' failed
make[2]: *** [lib/src/CMakeFiles/OT.dir/Base/Stat/HMatrixImplementation.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:831: recipe for target 'lib/src/CMakeFiles/OT.dir/all' failed
make[1]: *** [lib/src/CMakeFiles/OT.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
```

The guilty commit seems to be the following: https://github.com/openturns/openturns/commit/57b23f280ae449af6dbc542ae5a44cf267d3b0ec

The cmake command seems to find hmat:

```
$ cmake .. -DCMAKE_INSTALL_PREFIX=$PWD/install -DCOIN_ROOT_DIR=/home/osboxes/Documents/Bonmin-stable/build/install -DPYTHON_INCLUDE_PATH=/usr/include/python3.7m -DPYTHON_LIBRARIES=/usr/lib/python3 -DPYTHON_EXECUTABLE=/usr/bin/python3 -DUSE_SPHINX=ON -DSPHINX_FLAGS="-j10"
-- A library with BLAS API found.
-- A library with LAPACK API found.
-- Could NOT find Spectra (missing: SPECTRA_INCLUDE_DIRS) 
-- Boost version: 1.65.1
-- Could NOT find NLopt (missing: NLOPT_LIBRARY NLOPT_INCLUDE_DIR NLOPT_INCLUDE_DIRS NLOPT_LIBRARIES NLOPT_ROOT_DIR) 
-- Could NOT find dlib (missing: dlib_DIR)
-- Could NOT find OPTpp (missing: OPTPP_LIBRARY OPTPP_INCLUDE_DIR OPTPP_INCLUDE_DIRS OPTPP_ROOT_DIR) 
-- Found installed version of Eigen: /usr/lib/cmake/eigen3
-- Found required Ceres dependency: Eigen version 3.3.4 in /usr/include/eigen3
-- Found required Ceres dependency: glog
-- Found installed version of gflags: /usr/lib/x86_64-linux-gnu/cmake/gflags
-- Detected gflags version: 2.2.1
-- Found required Ceres dependency: gflags
-- Found Ceres version: 1.13.0 installed in: /usr with components: [LAPACK, SuiteSparse, SparseLinearAlgebraLibrary, CXSparse, SchurSpecializations, OpenMP]
-- Could NOT find COIN (missing: COIN_INCLUDE_DIR COIN_CBC_LIBRARY COIN_CBC_SOLVER_LIBRARY COIN_CGL_LIBRARY COIN_CLP_LIBRARY COIN_COIN_UTILS_LIBRARY COIN_OSI_LIBRARY COIN_OSI_CBC_LIBRARY COIN_OSI_CLP_LIBRARY COIN_BONMIN_LIBRARY COIN_IPOPT_LIBRARY) 
-- Could NOT find chaospy (missing: CHAOSPY_LOCATION _CHAOSPY_VERSION_MATCH) 
-- The following OPTIONAL packages have been found:

 * BISON (required version >= 2.4)
 * FLEX
 * TBB
 * muParser (required version >= 2.2.3)
 * HMAT (required version >= 1.2)
 * Doxygen
 * LibXml2
 * Boost (required version >= 1.46)
 * Ceres
 * CMinpack
 * SWIG (required version >= 2.0.9)
 * PythonInterp
 * PythonLibs (required version == 3.6.9)

-- The following REQUIRED packages have been found:

 * BLAS
 * LAPACK
 * Threads

-- The following OPTIONAL packages have not been found:

 * Spectra
 * NLopt
 * dlib
 * OPTpp
 * COIN

-- Configuring done
-- Generating done
-- Build files have been written to: /home/osboxes/GIT/openturns/build
```

The installed version of hmat is 1.2.0:
```
$ ls libhmat-oss.so*
libhmat-oss.so  libhmat-oss.so.1  libhmat-oss.so.1.2.0
```

**Comment by jschueller** [comment]
yes, pending fix in #1370

**Comment by mbaudin47** [comment]
Would you have a trick to build the master meanwhile ?
